### PR TITLE
chore: Add product update templates

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

> [!NOTE]
> This started out as PR templates, but we need to be able to create them ahead of code changes.

These have been broken up from the old template which has the following problems:
- It covers every product in one issue, meaning the issue stays open for a long time and progress is difficult to visualise on the engineering board.
- Concurrent updates to the issue description across the team is buggy (eg: old changes reverted and replaced by new changes. Sometimes the description becomes locked as a result.
- Improving instructions on the old template means the description becomes even longer.

This change:
- [x] Refactor the pre-release tracking issue with links to quickly create the issues from these templates.
- [x] Splits up the product update issue templates per product or product group for easier tracking on the engineering board.
- [x] Automatically adds issues to the engineering project board
- [x] There are specific instructions per-product to help the person doing the updates work autonomously. These aren't perfect, but can be improved over time.

To consider:
- [x] hello-world **Added in a tasklist in pre-release for now**
- [x] krb5 **Added in a tasklist in pre-release for now**
- [x] tools (and other misc items) **Added in a tasklist in pre-release for now**
- [ ] Moving ubi*-rust-builders out of the release process (they can be done as needed by developers or when SecObserve/Renovate recommend it).
- [ ] Revamping the existing PR template
- [x] Issue template for tasks when adding a new product/base

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/update-airflow.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+- [ ] _Link to any other operator PRs (getting_started / kuttl)_
+- [ ] _Link to demo PR (raise against the `next` branch)_
+```
+
+<!--
+Make this a regular list so it isn't easily editable from the rendered
+description?
+-->
+```[tasklist]
+### Acceptance
```
Comment (adwk67) [inline .github/ISSUE_TEMPLATE/update-airflow.md]: Nice to have: confirmation that it works on Openshift, too (although that could be arguably covered by the operator PR where we already have a "works on Openshift" check).

File: .github/ISSUE_TEMPLATE/update-hdfs.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# See the latest version at https://pypi.org/project/image-tools-stackabletech/
+pip install image-tools-stackabletech==0.0.12
+
+bake --product hadoop=x.y.z # where x.y.z is the new version added in this PR
+
+kind load docker-image docker.stackable.tech/stackable/hadoop:x.y.z-stackable0.0.0-dev
+
+# Change directory into the hdfs-operator repository and update the
+# product version in tests/test-definition.yaml
+./scripts/run-tests --test-suite smoke-latest
```
Comment (adwk67) [inline .github/ISSUE_TEMPLATE/update-hdfs.md]: Question: do all operators have `smoke-latest`?

File: .github/ISSUE_TEMPLATE/update-hive.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Update `versions.py` to the latest supported version of JVM (base and devel).
+- [ ] Update other dependencies if applicable (eg: jmx_exporter, aws_java_sdk_bundle, etc).
+- [ ] Check other operators (getting_started / kuttl) for usage of the versions. Add the PR(s) to the list below.
+- [ ] Update the version in demos. Add the PR(s) to the list below.
+```
+
+```[tasklist]
+### Related Pull Requests
+- [ ] _Link to the docker-images PR (product update)_
+- [ ] _Link to [hdfs-utils](https://github.com/stackabletech/hdfs-utils/) PR (if applicable)_
```
Comment (adwk67) [inline .github/ISSUE_TEMPLATE/update-hive.md]: Don't we need this rather in the hdfs template?

File: .github/ISSUE_TEMPLATE/update-java-bases.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+<details>
+<summary>Testing instructions</summary>
+
+```shell
+# See the latest version at https://pypi.org/project/image-tools-stackabletech/
+pip install image-tools-stackabletech==0.0.12
+
+# Test a product image can build, eg: ZooKeeper
+bake --product zookeeper=x.y.z # where x.y.z is a valid product vesion using the newly added Java version
```
Comment (adwk67) [inline .github/ISSUE_TEMPLATE/update-java-bases.md]:
```suggestion
bake --product zookeeper=x.y.z # where x.y.z is a valid product version using the newly added Java version
```

File: .github/ISSUE_TEMPLATE/update-opa.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+# See the latest version at https://pypi.org/project/image-tools-stackabletech/
+pip install image-tools-stackabletech==0.0.12
+
+bake --product opa=x.y.z # where x.y.z is the new version added in this PR
+
+kind load docker-image docker.stackable.tech/stackable/opa:x.y.z-stackable0.0.0-dev
+
+# Change directory into the opa-operator repository and update the
+# product version in tests/test-definition.yaml
+./scripts/run-tests --test-suite smoke-latest
```
Comment (adwk67) [inline .github/ISSUE_TEMPLATE/update-opa.md]: Question: do we have a test with rego rules? Maybe that would be better if the rules-version has changed.

File: .github/ISSUE_TEMPLATE/update-stackable-base.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+```shell
+# See the latest version at https://pypi.org/project/image-tools-stackabletech/
+pip install image-tools-stackabletech==0.0.12
+
+bake --product vector=x.y.z # where x.y.z is a valid version
+```
+
+</details>
+
+_Please consider updating this template if these instructions are wrong, or
```
Comment (adwk67) [inline .github/ISSUE_TEMPLATE/update-stackable-base.md]: :+1:

File: .github/ISSUE_TEMPLATE/update-zookeeper.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Update `versions.py` to the latest supported version of JVM (base and devel).
+- [ ] Update other dependencies if applicable (eg: jmx_exporter, etc).
+- [ ] Check other operators (getting_started / kuttl) for usage of the versions. Add the PR(s) to the list below.
+- [ ] Update the version in demos. Add the PR(s) to the list below.
+```
+
+```[tasklist]
+### Related Pull Requests
+- [ ] _Link to the docker-images PR (product update)_
+- [ ] _Link to [hdfs-utils](https://github.com/stackabletech/hdfs-utils/) PR (if applicable)_
```
Comment (adwk67) [inline .github/ISSUE_TEMPLATE/update-zookeeper.md]: Question: does zookeeper use hdfs-utils? As far as I know it contains the group-mapper, opa-authorizer and topology provider (all relevant to hdfs)

File: .github/ISSUE_TEMPLATE/update-airflow.md (hunk 1 above)
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/update-airflow.md]: I should have mentioned it, but I specifically left out comprehensive testing from these task lists (when it was a PR template). But now that it is a tracking issue, I could add that acceptance criteria back in.

~The instruction then would be to run OpenShift tests in Jenkins (and some instructions at the end for steps to doing that), right?~

---

> although that could be arguably covered by the operator PR where we already have a "works on Openshift" check

Yeah, I guess it is covered somewhere. Thinking more about it now, the goal for the product updates is to add in the new versions (whether or not they work - the containers are built and available). So I do feel the testing for that should be covered by the respective operator.

File: .github/ISSUE_TEMPLATE/update-hdfs.md (hunk 2 above)
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/update-hdfs.md]: The ones I actually checked did, so I assumed it was templates. I will go back and check all.

If they aren't:
- Update the instructions here
- Update the operator tests so they are consistent (then these instructions won't need updating if it ever changes in future).

File: .github/ISSUE_TEMPLATE/update-hive.md (hunk 3 above)
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/update-hive.md]: I think this was copy-pasta. Will check...

File: .github/ISSUE_TEMPLATE/update-airflow.md (hunk 1 above)
Comment (adwk67) [reply .github/ISSUE_TEMPLATE/update-airflow.md]: Yes, fair enough. Testing on Openshift with replicated should now not be any more involved than other platforms.

File: .github/ISSUE_TEMPLATE/update-zookeeper.md (hunk 7 above)
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/update-zookeeper.md]: I think this was some copy-pasta (same as the hive one). Will fix

File: .github/ISSUE_TEMPLATE/update-hdfs.md (hunk 2 above)
Comment (adwk67) [reply .github/ISSUE_TEMPLATE/update-hdfs.md]: Or: 

> ./scripts/run-tests --test-suite {smoke, smoke-latest or similar}

and leave it up to the developer.

File: .github/ISSUE_TEMPLATE/update-hive.md (hunk 3 above)
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/update-hive.md]: Fixed in 39dc3ae2b9d9c9953cfe84581e0f2650223db1b3

File: .github/ISSUE_TEMPLATE/update-opa.md (hunk 5 above)
Comment (NickLarsenNZ) [reply .github/ISSUE_TEMPLATE/update-opa.md]: I'm not familiar with the opa-operator tests. In the context of the `opa` version bump, what changes _rules-version_? Is it the bundle builder?

Comment (adwk67) [reply .github/ISSUE_TEMPLATE/update-opa.md]: Actually, having looked at the tests again I think the smoke test is fine, as there is a rudimentary rule that is called. I think I was getting mixed up with a Trino [change](https://github.com/stackabletech/trino-operator/pull/494/files) that used different rule sets.

**Review by adwk67 — APPROVED** [review]
LGTM!

**Pull request merged**